### PR TITLE
Tree Selection

### PR DESCRIPTION
**Pull request by stuartmscott**
### Description:
Triggers OnSelectionChanged in SetSelection, adds ClearSelection, and TreeNoSelection.

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [x] Public APIs match existing style.
- [ ] Any breaking changes have a deprecation path or have been discussed.

### CODE REVIEW

File: widget/tree_internal_test.go [hunk 1]
```diff
@@ -278,6 +278,35 @@ func TestTree_MinSize(t *testing.T) {
...
+	}
+
+	tree.SetSelection("A")
+	select {
+	case s := <-selection:
+		assert.Equal(t, "A", s)
+	case <-time.After(1 * time.Second):
+		assert.Fail(t, "Selection should have occured")
+	}
+
```
Comment (andydotxyz) [inline widget/tree_internal_test.go:300]: Can you check that the Selection() method returns the correct result after the selection is changed as well?

File: widget/tree.go [hunk 2]
```diff
@@ -68,6 +71,11 @@ func NewTreeWithStrings(data map[string][]string) (t *Tree) {
 	return
 }
 
+// ClearSelection clears the current selection.
```
Comment (andydotxyz) [inline widget/tree.go:74]: ClearSelection and SetSelection are here but no Selection() method

**Review by andydotxyz — CHANGES_REQUESTED** [review]
From the agreed API list I think you're missing `Tree.Selection() string`

File: widget/tree_internal_test.go (hunk 1 above)
Comment (stuartmscott) [reply widget/tree_internal_test.go:301]: Done!

File: widget/tree.go (hunk 2 above)
Comment (stuartmscott) [reply widget/tree.go:74]: Ooops forgot that one. Added now :)

**Review by andydotxyz — APPROVED** [review]

**Pull request merged**